### DISCUSSION TITLE
How to make files only viewable using iframe.

### DISCUSSION THREAD

**Discussion by schoolIsntFun** [community/community#78682] [Programming Help]
### Body

Hello, I have a github pages running hasoaxe.github.io . i have a problem, i have put a password system for fun, but some people just skip over it by typing the url. Hasoaxe.github.io/3kh0  or somthing. is there a way to make the file only viewable thru iframe which i have on the main page after you type the password. the repo is https://github.com/Hasoaxe/Hasoaxe.github.io   if anyone needs that.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by AranJannson** [comment]
Hi, dont know how helpful this might be but using your scripting language you may want to check for a flag that indicates wether a valid password has been entered, given the password is entered and valid. When you go to access a protected page it would allow you access whilst if theres no user logged in for say then you could either cause a redirect for the user to send them to an error page returning a 401 error or perhaps give an alert telling them they are unauthorized to access the page due to having not logged in.

From what i can see you just use a *index.html* file for the main page. If you where to perhaps use a web framework such as (React, Django, Nuxt...) then this functionality would be much easier to implement. For example in Django you could use the default User Model, as then you can check for a **@login_required** on the url in order to allow access to the page whilst further also adding inside of the protected page of choice 

The following: 

**{% if user.is_authenticated %}**  

**<-Your html->** 

**{% else %}**

**<-Html for unauthorized users->**  

This allows you to load specific html depeding on a users logged in status or not, now if you dont want to use a User model but still want a validation based on just a password you may want to look into creating a custom model which just takes a password as an encrypted field **(Most likely based on _SHA256_ if using the same encyprtion scheme as Django )** which can then be checked on its own with no *username* being required.

Hope my example in Django at least guides you in the right direction and or the first possible solution i provided in the first paragraph helps in understanding possible ways you could go about finding a solution.
